tests/src/Simple/Threading/Threading.cs

```diff
@@ -391,6 +394,25 @@ public static void SemaphoreTest()
     }
 
     [Fact]
+    public static void Reproduce3616()
+    {
+        AutoResetEvent resetEvent = new AutoResetEvent(false);
+        Thread thread = new Thread(() => {
+            var i = 0;
+            var tickCount = Environment.TickCount;
+            while(Environment.TickCount < tickCount + 1000) {
```

Comment:
Could just use `Thread.Sleep` instead. A one second sleep seems unnecessarily long and typically would not repro the issue. The two threads need to race in timing out waiting for the event and setting the event, so a better way may be to synchronize both threads with an event (say a `threadStarted` event that the main thread waits for and the background thread sets immediately upon starting). Then, the main thread can wait on `resetEvent` for say 50 ms and the background thread can sleep for 50 ms before setting the event twice.